### DISCUSSION TITLE
Change Chipset

### DISCUSSION THREAD

**Discussion by Chiliyago** [PlummersSoftwareLLC/NightDriverStrip#243] [Q&A]
When I attempt to upload my program to my board, I get the following error: A fatal error occurred: This chip is ESP32-S3 not ESP32

I am not sure where to change that value and have looked all over. Sorry for the basic question.

**Comment by davepl** [comment]
That’s a different chip so you have two options:

- Look in platformio.ini for one of the configurations that is intended for the S3, and run that, or
- Change the target of the project you’re working on to reflect the S3 by changing the “extends” line to be dev_esp32-s3

Note that the mesmerizer config won’t work, as the S3 does not support DMA, but the rest should work for you with that change.

- Dave

**Reply by Chiliyago** [reply]
Still failing miserably.  Is this what you meant by changing the extends?

[platformio]
default_envs = demo

I changed this:
[env:demo]
extends         = dev_esp32
build_flags     = -DDEMO=1
                  -Ofast
                  ${dev_esp32.build_flags}
 to that:
[env:demo]
extends         = dev_esp32-s3
build_flags     = -DDEMO=1
                  -Ofast
                  ${dev_esp32-s3.build_flags}

**Comment by davepl** [comment]
That looks like I thought it should… still refuses to flash?  Rutger knows perhaps a bit more about the dependency chain, I’ve added him to the mail, perhaps he can comment?

Cheers,
Dave

**Comment by Chiliyago** [comment]
If this helps and (I am not sure it does)... I can create a blank project with the following platformio.ini and it compiles and uploads to my device.

[env:esp32s3box]
platform = espressif32
board = esp32s3box
framework = arduino

**Comment by Chiliyago** [comment]
This is the board I have: https://www.amazon.com/dp/B076KJZ5QM?psc=1&ref=ppx_yo2ov_dt_b_product_details

**Comment by davepl** [comment]
Ah… are you sure that’s an S3?  V3 != S3

The V3 isn’t yet supported by PlatformIO, as I understand it, but the change to enable it is supposedly coming soon from Espressif.

Here’s a thread on it:  https://github.com/platformio/platform-espressif32/issues/946

Have you tried just putting the “code” you posted into the platformio.ini for a Nightdriver config?  I don’t see why that wouldn’t work!

[env:esp32s3box]
platform = espressif32
board = esp32s3box
framework = arduino

- Dave

**Comment by Chiliyago** [comment]
I'd like to but not yet entirely sure the proper way to do so.  I'd am not fully clear about the concepts of structuring the ini file in this project but when I do, I'll try to add some additional documentation for you to pull.
